### PR TITLE
Provision CI from conda-forge

### PR DESCRIPTION
**Pull request by awvwgk**
Using conda environments to provision the dependencies for CI testing seems more stable than using the base images.

### CODE REVIEW

**Comment by marvinfriede** [comment]
I also fixed the macos build (see [here](https://github.com/dftd4/dftd4/blob/e6f8b46d7685b5c1b079b342b6720216a1c6c9b3/.github/workflows/fortran-build.yml#L52)).

**Comment by awvwgk** [comment]
The CI breakage over time is quite annoying, I found that repairing it in [dftd3/simple-dftd3] was far easier than here. Therefore, I copied over the CI from there and adapted it to match the additional requirements (BLAS/Lapack). Feel free to decide which variant you prefer.

File: .github/workflows/fortran-build.yml [hunk 1]
```diff
@@ -3,201 +3,287 @@ name: CI
...
-      GCC_V: ${{ matrix.gcc_v }}
-      OMP_NUM_THREADS: 2,1
+      FC: ${{ matrix.compiler == 'intel' && 'ifort' || 'gfortran' }}
+      CC: ${{ matrix.compiler == 'intel' && 'icc' || 'gcc' }}
+      GCC_V: ${{ matrix.version }}
       PYTHON_V: 3.8
 
     steps:
     - name: Checkout code
       uses: actions/checkout@v2
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:69]:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/fortran-build.yml [hunk 2]
```diff
@@ -3,201 +3,287 @@ name: CI
...
+    - name: Prepare for cache restore
+      if: ${{ matrix.compiler == 'intel' }}
+      run: |
+        sudo mkdir -p /opt/intel
+        sudo chown $USER /opt/intel
+
+    - name: Cache Intel install
+      if: ${{ matrix.compiler == 'intel' }}
+      id: cache-install
+      uses: actions/cache@v2
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:120]:
```suggestion
      uses: actions/cache@v3
```

File: .github/workflows/fortran-build.yml [hunk 3]
```diff
@@ -3,201 +3,287 @@ name: CI
...
         tar cvf ${{ env.OUTPUT }} _dist
         xz -T0 ${{ env.OUTPUT }}
         echo "DFTD4_OUTPUT=${{ env.OUTPUT }}.xz" >> $GITHUB_ENV
       env:
-        OUTPUT: dftd4-gcc-${{ matrix.gcc_v }}-${{ matrix.os }}.tar
+        OUTPUT: dftd4-${{ matrix.compiler }}-${{ matrix.version }}-${{ matrix.os }}.tar
 
     - name: Upload package
+      if: ${{ matrix.build == 'meson' && matrix.build-type != 'coverage' }}
       uses: actions/upload-artifact@v2
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:227]:
```suggestion
      uses: actions/upload-artifact@v3
```

File: .github/workflows/fortran-build.yml [hunk 4]
```diff
@@ -3,201 +3,287 @@ name: CI
...
       CC: gcc
       GCC_V: ${{ matrix.gcc_v }}
       PYTHON_V: ${{ matrix.python_v }}
       OMP_NUM_THREADS: 2,1
-      DFTD4_OUTPUT: dftd4-gcc-${{ matrix.gcc_v }}-${{ matrix.os }}.tar.xz
+      DFTD4_OUTPUT: dftd4-gnu-${{ matrix.gcc_v }}-${{ matrix.os }}.tar.xz
 
     steps:
     - name: Checkout code
       uses: actions/checkout@v2
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:261]:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/fortran-build.yml [hunk 5]
```diff
@@ -3,201 +3,287 @@ name: CI
...
         sudo update-alternatives
         --install /usr/bin/gcc gcc /usr/bin/gcc-${{ env.GCC_V }} 100
         --slave /usr/bin/gfortran gfortran /usr/bin/gfortran-${{ env.GCC_V }}
         --slave /usr/bin/gcov gcov /usr/bin/gcov-${{ env.GCC_V }}
 
-    - name: Install meson and test dependencies
-      run: pip3 install ${{ env.PIP_EXTRAS }}
-
     - name: Download package
       uses: actions/download-artifact@v2
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:288]:
```suggestion
      uses: actions/download-artifact@v3
```

File: .github/workflows/fortran-build.yml [hunk 6]
```diff
@@ -222,238 +308,9 @@ jobs:
       run: pytest --pyargs dftd4 --cov=dftd4 -vv
       env:
         LD_LIBRARY_PATH: ${{ env.LD_LIBRARY_PATH }}:${{ env.DFTD4_PREFIX }}/lib
+        DYLD_LIBRARY_PATH: ${{ env.DYLD_LIBRARY_PATH }}:${{ env.DFTD4_PREFIX }}/lib
+
+    - run: pip3 install coverage
 
     - name: Upload coverage report
       uses: codecov/codecov-action@v1
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:316]:
```suggestion
      uses: codecov/codecov-action@v3
```

File: .github/workflows/fortran-build.yml [hunk 7]
```diff
@@ -3,201 +3,287 @@ name: CI
...
 
-    - name: Test Python API
-      run: pytest --pyargs dftd4 --cov=dftd4 -vv
-      env:
-        LD_LIBRARY_PATH: ${{ env.LD_LIBRARY_PATH }}:${{ env.DFTD4_PREFIX }}/lib
-        PYTHONPATH: ${{ env.PYTHONPATH }}:${{ env.DFTD4_PREFIX }}/lib/python${{ env.PYTHON_V }}/site-packages
-
     - name: Upload coverage report
+      if: ${{ matrix.build-type == 'coverage' }}
       uses: codecov/codecov-action@v1
```
Comment (marvinfriede) [inline .github/workflows/fortran-build.yml:234]:
```suggestion
      uses: codecov/codecov-action@v3
```

**Review by marvinfriede — APPROVED** [review]

**Pull request merged**